lab_1_keywords_tfidf/main.py

```diff
@@ -118,6 +200,38 @@ def calculate_expected_frequency(
 
     In case of corrupt input arguments, None is returned
     """
+    if isinstance(doc_freqs, dict) is False or isinstance(corpus_freqs, dict) is False:
+        return None
+    else:
+        for k, v in doc_freqs.items():
+            if isinstance(k, str) is False or isinstance(v, int) is False:
+                return None
+        for k, v in corpus_freqs.items():
+            if isinstance(k, str) is False or isinstance(v, int) is False:
+                return None
+        # подумать как решить это по другому
+        words_in_doc = 0
+        words_in_col = 0
+        doc_freqs_val = list(doc_freqs.values())
+        col_freqs_val = list(doc_freqs.values())
+        for i in doc_freqs_val:
+            words_in_doc += i
+        for i in col_freqs_val:
+            words_in_col += i
+        exp_freqs = {}
+        for key, val in corpus_freqs.items():
+            for i in range(doc_freqs_val):
+                l = words_in_doc - doc_freqs_val[i]
+                # l -  количество вхождений всех слов, кроме , в документ d
+                m = words_in_col - col_freqs_val[i]
+                # m - количество вхождений всех слов, кроме , в коллекцию документов D
+                j = doc_freqs_val[i]
+                # j - количество вхождений слова t  в документ d
+                k = col_freqs_val[i]
+                # k - количество вхождений слова t во все тексты коллекции D
+                exp = ((j + k) * (j * l))/ (j + k + l + m)
+                exp_freqs[key] = exp
```

Comment:
let's examine what happens here

you iterate over each key of in corpus_freqs. 
at each iteration, you also perform a cycle over [0; number of values in doc_freqs - 1]
each time a `i` takes a new value from the range from 0 to the (number of values in doc_freqs - 1), you compute `k`, `l`, `j`, `m`, `exp` and (again) add a pair to the dictionary. when addind a pair with the same key to the dict, it overwrites the previous one as dictionaries can only have unique keys 

consider iterating over keys of one dictionary (which one?), requesting the value that corresponds to that key from another dictionary (you could use get here like you did earlier) and working with these values